### PR TITLE
support kernel 5.12+

### PR DESCRIPTION
**Pull request by dimorinny**
Signature changes, that allow using simplefs with 5.12+ kernel.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Use conditional build to allow older kernel versions.

**Comment by dimorinny** [comment]
Done

File: simplefs.h [hunk 1]
```diff
@@ -16,6 +16,10 @@
 #define SIMPLEFS_FILENAME_LEN 28
 #define SIMPLEFS_MAX_SUBFILES 128
 
+#include <linux/version.h>
+
+#define user_namespace_required() LINUX_VERSION_CODE >= KERNEL_VERSION(5,12,0)
```
Comment (jserv) [inline simplefs.h]: Use capitalized macro. That is, `USER_NS_REQUIRED`.

**Comment by jserv** [comment]
Thank @dimorinny for contributing!

**Pull request merged**